### DISCUSSION TITLE
max_vertices/max_primitives/local_size_x size tuning

### DISCUSSION THREAD

**Discussion by Try** [zeux/niagara#32] [Q&A] [answered]
Hi, I have a question: what are good numbers for `max_vertices` / `max_primitives`, from you experience ?
https://github.com/zeux/niagara/blob/e9f3a890af764d765efd544e243452c380b08d50/src/shaders/meshlet.mesh.glsl#L16-L17

Potentially with such layout driver will spawn ~64 helper threads, that wont do any real work: [see slide 24](https://indico.freedesktop.org/event/2/contributions/49/attachments/55/80/xdc2022_mesa_mesh_shader_implementation.pdf)
Also on stream, I've noticed that you are using Linux+AMD, so mesa guideline is probably very relevant to you machine.

On my side, after EXT came out, I've changed shader in my engine to `local_size_x=max_primitives=max_vertices=64` - seems to work well on NV, and can't test on AMD unfortunately.

Can you test `max_primitives=max_vertices=64` layout? Thanks!

**Comment by zeux** [comment, marked as answer]
As I mentioned on stream, we’re not using an efficient configuration right now. From my brief experiments 64/64 right now seems to be optimal on radv but it feels like some other options may be viable as well. We’ll explore that in a future stream.
